File: leetcode/119. Pascal's Triangle II.cpp
class Solution {
public:
     
    vector<int> getRow(int rowIndex) {
    
        vector<int> result(rowIndex + 1);
        result[0] = result[rowIndex] = 1;
        int size = result.size();
        for(int i = 1 ; i < (size+1)/2 ; i++){
            result[i] = result[size-i - 1] = ((long)result[i-1]*(size - i))/i;
        }
        return result;    
    }
};

File: online-training-platform/leetcode/119. Pascal's Triangle II.kt
class Solution {
    fun getRow(rowIndex: Int): List<Int> {
        return when(rowIndex) {
            0 -> listOf(1)
            1 -> listOf(1,1)
            2 -> listOf(1,2,1)
            else -> {
                var current = getRow(2)
                var result = mutableListOf(1)
                for(idx in 2 until rowIndex){
                    result = mutableListOf(1)
                    for(j in 1 until current.size) {
                        result.add(current[j] + current[j - 1])
                    }
                    result.add(1)
                    current = result
                }
                result
            }
        }
    }
}